### ISSUE TITLE
rclone MUST be reinstalled after reboot, remount does not work

### ISSUE THREAD

**Issue by HASJ** [Magisk-Modules-Repo/com.piyushgarg.rclone#23]
I do not know the cause. Everything else works fine.

**Comment by cavanist** [comment]
Same here
Samsung Galaxy Tab S2
Android 6.0.1 Stock
Magisk 19.3
rclone 1.7
 
works only after instalation till reboot...

**Comment by Geofferey** [comment]
@HASJ @cavanist  Does `rclone remount` work from terminal on reboot or no? What are the results of running that command? This might be due to the new `PATH=` variable at top of `service.sh` which tells the script where to find all the system bins. Could you try commenting out that line `#` and re-running for me? Also, are your devices encrypted by any chance?

**Comment by cavanist** [comment]
I can share my eperience, hope I'dont make more troubles than explanation :-)
So according to my needs I have three completely same devices. The first is Lineafe 16 based on android pie - where i have no trouble to set up rclone module... just put confing in .rclone folder, install magisk module and that's it. Works very realiable when it boots up, using mpv and vlc to play media blazing fast ;-)
the second device is based on original stock rom based on android 6 ( it's more responsible than 5 or 7 stock roms on this device) and yes this device is encrypted since the first boot by manufacturer... the third device is based on the same rom but i did some magic before first start, following instructions on xda, so it has stock rom but it's not encrypted. I haven't tried it yet if rclone works on this third device.
so 
first device - working rclone module
second device - the mounts are lost after reboot device, and the mounted folder are empty!!
third device - not tested yet

I would share with you command outputs, but I'm little bit confused there. The magisk rclone module works itself independently? Because I haven't install rclone from repository on any of this device, simply I wasn't able to use any rclone command from terminal,  because it wasnt't there......but Magisk module works perfectly (at least on the first device i have described) During the problems on second device  I have downloaded and install official rclone package to my systems - the remount command is not recognized at all. So this is my experience.

**Comment by Geofferey** [comment]
@cavanist Yes the `rclone` command should work inside of a terminal emulator on your device. Try `rclone remount` on phone 2 and see if it runs or says `sh: rclone: not found`. If that happens I know binaries aren't being linked at boot. If it works it's probably some check I have implemented. Try `NETCHK=0` in` .global.param` as well so I can rule that out. ;)

**Comment by cavanist** [comment]
So these results:
rclone available commands
https://s18.directupload.net/images/190820/elctlx7t.jpg

`rclone remount` leads to `sh: rclone: not found`, added `NETCHK=0` in `.global.param`
there were no any log files in the folder after instalation, and no log files after reboot... the mounted folder before reboot was empty.
here is what i got after module instalation https://s18.directupload.net/images/190820/gcdqyvbt.jpg

I have tried to run `rclone remount` command on the first device where the module works without problems and it returns the same `sh: rclone: not found` ;-))

I have joined since the version 1.8 few days ago :-)

**Comment by Geofferey** [comment]
@cavanist So just to be clear when you install module initially on second dev you get empty folders? On reboot you get nothing? Have you tried the `rclone remount` command after a reboot. 

What are results of following command:

`getprop service.bootanim.exit`

**Comment by cavanist** [comment]
yes yes - `nmt\cloud\synologyx` is empty, there are no files on this problematic second device... 
when i reboot the device `cloud` folder is no more no presented in `nmt` folder
the result of `getprop service.bootanim.exit` is `0`
i have tried `rclone remount` on the both devices many times without succes....

**Comment by cavanist** [comment]
https://s18.directupload.net/images/190820/x8d4szyx.jpg

**Comment by Geofferey** [comment]
@cavanist Okay so one problem is the use of Termux (maybe mistake but I recognize font). Please try another emulator as Termux has it's own env and `PATH=` which  causing it to utilize the binaries installed in Termux. I recommend personally [JuiceSSH] for it's arrow keys etc (https://play.google.com/store/apps/details?id=com.sonelli.juicessh) or you can also run `export PATH=/system/bin:/sbin:/sbin/.core/busybox:/system/xbin` inside Termux to change it's used binaries. Try the `rclone remount` command then and let me know if it outputs like Magisk.

**Comment by cavanist** [comment]
Okay - it seems a have more information right now. ;-) I can confirm it's termux behavior, but at the same time using it co configure some other modules like LKT, which has beautiful terminal enviroment  :) lol.. It's no problem me to switch. 
Let's back to our case. For this moment I did run `rclone remount` from this command line app
https://play.google.com/store/apps/details?id=com.myapkapp.surootcommand

and i got:

https://s18.directupload.net/images/190820/kk2dzwi5.jpg

thanks for your guindances, hope it brings something important

**Comment by Geofferey** [comment]
@cavanist Your mount populated as well? If you're comfortable you can make a small edit to the `service.sh` located in `/sbin/.core/img/com.piyushgarg.rclone/`. At line `412` (in my editor) remove `&& [[ $(getprop service.bootanim.exit) = 1`. You could also use a find to search for line :). Should fix and hold you off until I publish a verified fix.

**Comment by cavanist** [comment]
Okay okay
 I did removal of 
<img src="https://s18.directupload.net/images/190821/weyurvz9.jpg">

Since this edit  there is no more problem after reboot ;) but the mounted folder still keeps empty

**Comment by Geofferey** [comment]
@cavanist What does `rclone.log` in `/sdcard/.rclone/` say? The last few entries should be enough.

**Comment by cavanist** [comment]
Another strange thing I never had `rclone.log` presented in `/sdcard/.rclone/` folder.. even if i try `rclone remout` or if i reboot, the log file was never there... i just try to push it, so i created `.synologyx.param` with `LOGFILE=/sdcard/.rclone/rclone. log` but it doesn't help...
<img src="https://s18.directupload.net/images/190821/2qsmhxuf.jpg">

**Comment by Geofferey** [comment]
@cavanist One last thing before I say this is out of my expertise.. Have you tried to comment out `PATH=` in the `service.sh`? I think I know what's going on... It's because of the `unused DT entry: type 0x1d arg 0x46d` what I'm not sure of is if it's coming from `rclone` `fusermount` or some other binary being used, but looks as if it's rclone. What happens when you run `rclone listremotes --config  /sdcard/.rclone/rclone.conf`

Try mounting manually?
`rclone mount synologyx: /mnt/cloud/synologyx --config /sdcard/.rclone/rclone.conf`

Also see:
https://stackoverflow.com/questions/33206409/unused-dt-entry-type-0x1d-arg

I'm gonna talk to someone once I find out if its `rclone` or `fusermount`. Unfortunately I can't guarantee a fix at this point. I won't stop trying but I don't expect it to come fast if I gotta compile stuff.

**Comment by cavanist** [comment]
What i've got:
`rclone listremotes --config /sdcard/.rclone/rclone.conf`
<img src="https://s18.directupload.net/images/190822/rchlhdja.jpg">
`rclone mount synologyx: /mnt/cloud/synologyx --config /sdcard/.rclone/rclone.conf`
<img src="https://s18.directupload.net/images/190822/o5wtrdlo.jpg">

I also tried to delete paths one by one  from `PATH=` to change priority and then perform `rclone remout`, but folder is empty still...

**Comment by Geofferey** [comment]
@cavanist I have an Idea... Try those same commands in su Termux and see if they work. If so maybe we can fix this ;) If you want we can establish another line of communication to help speed up this process.

**Comment by cavanist** [comment]
yes, i would prefer that, because it starts to be little geeky to me :-)

**Comment by Geofferey** [comment]
@cavnist in Termux `pkg install rclone` take the `rclone` bin from `/data/data/com.termux/usr/bin/` and place it in `/sbin/.core/img/com.piyushgarg.rclone/`... I think that will get you up and running.

**Comment by cavanist** [comment]
I did try on device 2 & 3, (both stock, encrypted & unencrypted)
I've got from Termux
<img src="https://s18.directupload.net/images/190822/e4cynhrl.jpg">
From JuicySSH  various results, you shloud have to see it... on device 2 unused DT entry dissmised, on device 3 I've got some new things...

please checkout your facebook messenger for my request :)

**Comment by mke2fs** [comment]
I have similar issue. 

I ran rclone remount from a terminal and it said done. But after a few seconds "Permission denied" popped up. And there is nothing showed inside the mount point.

Any idea?

**Comment by Geofferey** [comment]
@mke2fs Did you run as root? Can you send screenshots? Did it work on install?

**Comment by mke2fs** [comment]
I did run it as root.

![IMG_20190823_074039](https://user-images.githubusercontent.com/54405165/63557386-130d7400-c538-11e9-8bd1-842276cc11d4.jpg)

Phone Model: oneplus 3T
Android Version: 9.0
Module Ver: 1.48.0 mod 1.7 (107) 
Magisk Ver: 19.4 (19307)

**Comment by piyushgarg** [comment]
check the permission on your fusermount and rclone binaries.
ls -l /sbin/.core/modules/com.piyushgarg.rclone
..
-rwxr-xr-x 1 root root    55888 2019-08-09 18:58 fusermount
-rwxr-xr-x 1 root root      306 2019-08-09 18:58 fusermount-wrapper.sh
-rwxr-xr-x 1 root root 52970472 2019-08-09 18:58 rclone
-r-x------ 1 root root     4361 2019-08-09 18:58 rclone-wrapper.sh
..
else I wonder if the stock kernel supports fuse.

**Comment by mke2fs** [comment]
They all look good.

![IMG_20190823_145524](https://user-images.githubusercontent.com/54405165/63572728-1d992f00-c573-11e9-8652-89b23b96d0c5.jpg)

I am not running stock kernel but a LineageOS based custom rom. It looks like the kernel support fuse per output of cat /proc/filesystems.

**Comment by mke2fs** [comment]
Update:  It looks like the problem is with the service.sh script. If I manually run the whole rclone mount commandline as found inside the script (i.e. su -M ........), it works without issue.

**Comment by mke2fs** [comment]
Update2: OK. I missed those RCLONE_PARAMS in my previous run. Here is the screenshot of service.sh (I wrapped that rclone mount line with set-x and set +x).

![IMG_20190823_211546](https://user-images.githubusercontent.com/54405165/63595480-98307180-c5a8-11e9-94c0-8349bd118ad0.jpg)

**Comment by Z-hadum** [comment]
> 
> 
> @HASJ @cavanist Does `rclone remount` work from terminal on reboot or no? What are the results of running that command? This might be due to the new `PATH=` variable at top of `service.sh` which tells the script where to find all the system bins. Could you try commenting out that line using a `#` and re-running for me? Also, are your devices encrypted by any chance?

Hi.
so I have a similar problem (mounts available directly after installation but lost a restart). I have tried  `rclone remount`  via JuiceSSH and get  `bash: rclone: command not found`. Also tried adding a file `.global.params` in `/sdcard/.rclone/`  and commenting out the PATH line in `/data/adb/.../service.sh` - no change. I deleted `/sdcard/.rclone/rcolone.log` (it was 0 byte) and this file does not reapear after reboot. 

System: Android 9 LineageOS16.0.20200327-UNOFFICIAL-flo with Magisk 20.4 and "rclone-mount v1.50.2 mod v1.8."

Any idea what to do?

**Comment by spark1714** [comment]
> Hi.
> so I have a similar problem (mounts available directly after installation but lost a restart). I have tried `rclone remount` via JuiceSSH and got `bash: rclone: command not found`. Also tried adding the file `.global.params` with `NETCHK=0` in `/sdcard/.rclone/` and commenting out the PATH line in `/data/adb/.../service.sh` - no change. I deleted `/sdcard/.rclone/rcolone.log` (it was 0 byte) and this file does not reapear after reboot.
> 
I have exactly the same problem (oneplus5).
Were you able to solve it?
thx

**Comment by Z-hadum** [comment]
> 
> 
> > Hi.
> > so I have a similar problem (mounts available directly after installation but lost a restart). I have tried `rclone remount` via JuiceSSH and got `bash: rclone: command not found`. Also tried adding the file `.global.params` with `NETCHK=0` in `/sdcard/.rclone/` and commenting out the PATH line in `/data/adb/.../service.sh` - no change. I deleted `/sdcard/.rclone/rcolone.log` (it was 0 byte) and this file does not reapear after reboot.
> 
> I have exactly the same problem (oneplus5).
> Were you able to solve it?
> thx

Unfortunately: no...
